### PR TITLE
Stub build

### PR DESCRIPTION
**Pull request by bartoszglow**
Introduces additional command option for template builder:
```aem-site-template-builder stub```

New functionality builds a stub of site template which is a minimalistic skeleton. This includes Site Templates:
- package.json
- theme-sources.zip
- theme.zip (cleared out compiled theme output)

Theme sources include only:
- package.json (with theme info, npm scripts and npm dependencies)
- README.md 
- src/theme.scss (entry point for all styles)
- src/theme.ts (entry point for all js)

### CODE REVIEW

**Review by msagolj — APPROVED** [review]
Reviewed by testing stub command and using proxy command

File: bin/index.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env node
 
 /*
-Copyright 2021 Adobe
+Copyright 2020 Adobe
```
Comment (jckautzmann) [inline bin/index.js]: Why should the date be changed?

File: bin/index.js [hunk 2]
```diff
@@ -10,7 +10,16 @@ it.
...
-  await packageSiteTemplate({ rootPath: process.cwd() });
-})();
+const cmd = yargs
+  .scriptName('aem-site-template-builder')
+  .command(['serve', '$0'], 'build', () => {}, async function () {
+    await packageSiteTemplate({ rootPath: process.cwd() });
+  })
+  .command('stub', 'stub', () => {}, async function () {
+    await packageSiteTemplate({ rootPath: process.cwd(), stub: true });
+    // packageStub({ rootPath: process.cwd() });
```
Comment (comanV) [inline bin/index.js]: Is this line left commented on purpose?

File: bin/index.js [hunk 3]
```diff
@@ -10,7 +10,16 @@ it.
...
+  })
+  // .help()
```
Comment (comanV) [inline bin/index.js]: Same question as above.

**Review by comanV — APPROVED** [review]
Tested the new command for creating minimalistic site template and also its usage on a site created from archetype. 
I think it would be useful to document it in the README file.

**Comment by jckautzmann** [comment]
when using `theme.zip` and `theme-sources.zip` from the stub site template for the `Enable Front End Pipeline for existing sites` feature and running `npm run live`, I get following error:
```
Error: ENOENT: no such file or directory, stat 'dist'
    at Object.statSync (fs.js:982:3)
    at listFiles (/Users/kautzmann/Downloads/archetype1-theme-sources/node_modules/@adobe/aem-site-theme-builder/lib/listFiles.js:16:20)
    at /Users/kautzmann/Downloads/archetype1-theme-sources/node_modules/@adobe/aem-site-theme-builder/lib/proxy.js:111:21
    at call (/Users/kautzmann/Downloads/archetype1-theme-sources/node_modules/connect/index.js:239:7)
    at next (/Users/kautzmann/Downloads/archetype1-theme-sources/node_modules/connect/index.js:183:5)
    at Function.handle (/Users/kautzmann/Downloads/archetype1-theme-sources/node_modules/connect/index.js:186:3)
    at Server.app (/Users/kautzmann/Downloads/archetype1-theme-sources/node_modules/connect/index.js:51:37)
    at Server.emit (events.js:315:20)
    at parserOnIncoming (_http_server.js:789:12)
    at HTTPParser.parserOnHeadersComplete (_http_common.js:116:17)
```

**Comment by jckautzmann** [comment]
The content of the `package.json` of the downloaded `theme-sources.zip` should be updated with the information from the stub site template, not the standard site template as it is now:
```
{
  "name": "aem-site-template-standard",
  "version": "1.1.0",
  "description": "Standard AEM Site Template used as a starter template for general purpose websites",
  "license": "MIT License, Copyright 2020 Adobe Systems Incorporated",
  "scripts": {
    "build": "aem-site-template-builder",
    "deploy": "curl -u admin:admin -F file=@aem-site-template-standard-latest.zip http://localhost:4502/conf/global/site-templates.import.html"
  },
  "devDependencies": {
    "@adobe/aem-site-template-builder": "2.2.2"
  },
  "siteTemplate": {
    "title": "Standard AEM Site Template",
    "createdBy": "Adobe",
    "useCases": [
      "General purpose sites, corporate marketing sites, landing pages, blog sites, content marketing"
    ],
    "docsUrl": "",
    "showcaseUrl": "",
    "sourceRepositoryUrl": "https://github.com/adobe/aem-site-template-standard",
    "openItem": "{Path}/en/home"
  }
}

```

**Comment by jckautzmann** [comment]
The content of the README of the downloaded `theme-sources.zip` should be updated with the information from the stub site template, not the standard site template as it is now:
```
# Site Theme

This is the theme of the standard site template for Adobe Experience Manager (AEM).

This theme can be modified to customize the visual appearance of sites created from the standard site template.

## Structure

* `src/main.ts`: This is the main entry point of your JS & CSS theme.
* `src/site`: Files that are generic to the entire site.
* `src/components`: Files that are specific to components.
* `src/resources`: Associated files, like icons, logos, fonts.

## Build

1. Initialize the project with following command executed at the theme root:

```
npm install
```

2. Complete the `.env` file with credentials for the local proxy server to access the site created on Cloud Service.

3. Run the local proxy server while working to preview your changes with the content from the production environment.

```
npm run live
```

4. Once your work completed, check your changes into GitHub, and execute the deployment action on GitHub.

```

**Comment by comanV** [comment]
> when using `theme.zip` and `theme-sources.zip` from the stub site template for the `Enable Front End Pipeline for existing sites` feature and running `npm run live`, I get following error:
> 
> ```
> Error: ENOENT: no such file or directory, stat 'dist'
>     at Object.statSync (fs.js:982:3)
>     at listFiles (/Users/kautzmann/Downloads/archetype1-theme-sources/node_modules/@adobe/aem-site-theme-builder/lib/listFiles.js:16:20)
>     at /Users/kautzmann/Downloads/archetype1-theme-sources/node_modules/@adobe/aem-site-theme-builder/lib/proxy.js:111:21
>     at call (/Users/kautzmann/Downloads/archetype1-theme-sources/node_modules/connect/index.js:239:7)
>     at next (/Users/kautzmann/Downloads/archetype1-theme-sources/node_modules/connect/index.js:183:5)
>     at Function.handle (/Users/kautzmann/Downloads/archetype1-theme-sources/node_modules/connect/index.js:186:3)
>     at Server.app (/Users/kautzmann/Downloads/archetype1-theme-sources/node_modules/connect/index.js:51:37)
>     at Server.emit (events.js:315:20)
>     at parserOnIncoming (_http_server.js:789:12)
>     at HTTPParser.parserOnHeadersComplete (_http_common.js:116:17)
> ```

I had it working for me with "@adobe/aem-site-theme-builder" : "5.2.1".

**Comment by jckautzmann** [comment]
@comanV 
> I had it working for me with "@adobe/aem-site-theme-builder" : "5.2.1".

I used the `aem-site-theme-builder` from this branch.

**Comment by comanV** [comment]
@jckautzmann 
> I used the `aem-site-theme-builder` from this branch.

I agree, just saying that it should work after it will be updated.

**Pull request merged**